datasetIngestor/checkMetadata.go

```diff
@@ -193,18 +201,23 @@ func getHost() string {
 	}
 
 	for _, addr := range addrs {
-		if ipv4 := addr.To4(); ipv4 != nil {
-			ip, err := ipv4.MarshalText()
-			if err != nil {
-				return unknown
-			}
-			hosts, err := net.LookupAddr(string(ip))
-			if err != nil || len(hosts) == 0 {
-				return unknown
-			}
-			fqdn := hosts[0]
-			return strings.TrimSuffix(fqdn, ".") // return fqdn without trailing dot
+		ipv4 := addr.To4()
+		if ipv4 == nil {
+			continue
+		}
+		ip, err := ipv4.MarshalText()
+		if err != nil {
+			continue
+		}
+		hosts, err := net.LookupAddr(string(ip))
+		if err != nil || len(hosts) == 0 {
+			continue
+		}
+		fqdn := strings.TrimSuffix(hosts[0], ".") // fqdn without trailing dot
+		if !isValidDomain(fqdn) {
```

Comment:
the hostname returned by the command is the computer's advertised network name (if it can't find an FQDN).  If it's acceptable to pass that as a hostname, even though it's not an FQDN, it should be proposed to the SciCat dev team. That would probably imply that any string should be accepted as a hostname

Reply:
I have checked in our data, it's sometimes an alias which is not a FQDN. But I think it's OK like this for now, we could update our alias map

Reply:
I will add this to the issue comment